### PR TITLE
upgrade to cosmwasm 1.1.5

### PR DESCRIPTION
**Pull request by kwtalley**
This PR upgrades the CosmWasm dependencies to v1.1.2

### CODE REVIEW

File: contracts/scope/src/msg.rs [hunk 1]
```diff
@@ -3,19 +3,19 @@ use provwasm_std::Scope;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline contracts/scope/src/msg.rs:6]: Same comments as attrs/src/msg.rs, including the ones for QueryMsg.

File: contracts/tutorial/Cargo.toml [hunk 2]
```diff
@@ -38,12 +38,12 @@ backtraces = []
...
+cosmwasm-std = { version = "1.1.2", default-features = false }
+cosmwasm-storage = { version = "1.1.2" }
 schemars = "0.8.3"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.26" }
 
 [dev-dependencies]
 provwasm-mocks = { path = "../../packages/mocks" }
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
Comment (Taztingo) [inline contracts/tutorial/Cargo.toml:49]: Same comments as attrs/Cargo.toml.

File: contracts/tutorial/src/msg.rs [hunk 3]
```diff
@@ -5,7 +5,7 @@ use serde::{Deserialize, Serialize};
 use crate::state::State;
 
 /// A message sent to initialize the contract state.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline contracts/tutorial/src/msg.rs:8]: Same comments as the ones for attrs/src/msg.rs, including the ones for QueryMsg.

File: packages/bindings/Cargo.toml [hunk 4]
```diff
@@ -13,9 +13,9 @@ categories = ["api-bindings", "cryptography::cryptocurrencies", "wasm"]
...
 
 [dependencies]
-cosmwasm-std = { version = "1.0.0", default-features = false }
+cosmwasm-std = { version = "1.1.2", default-features = false }
 schemars = "0.8.1"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 
 [dev-dependencies]
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
Comment (Taztingo) [inline packages/bindings/Cargo.toml:21]: Same comments as attrs/Cargo.toml.

File: packages/bindings/src/msg.rs [hunk 5]
```diff
@@ -12,7 +12,7 @@ use crate::Scope;
 static MSG_DATAFMT_VERSION: &str = "2.0.0";
 
 /// Represents a request to encode custom provenance messages.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline packages/bindings/src/msg.rs:15]: Same comments as attrs/src/msg.rs.

File: packages/bindings/src/querier.rs [hunk 6]
```diff
@@ -154,7 +154,7 @@ impl<'a> ProvenanceQuerier<'a> {
     /// }
     ///
     /// // Text with timestamp.
-    /// #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+    /// #[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline packages/bindings/src/querier.rs:157]: Comment is out of date, should use the new derive.

File: packages/bindings/src/query.rs [hunk 7]
```diff
@@ -5,7 +5,7 @@ use serde::{Deserialize, Serialize};
 use crate::types::ProvenanceRoute;
 
 /// Represents a request to query a custom provenance module.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline packages/bindings/src/query.rs:8]: Same comments as attrs/src/msg.rs.

File: packages/bindings/src/types.rs [hunk 8]
```diff
@@ -3,7 +3,7 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 /// Supported provenance module router keys.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline packages/bindings/src/types.rs:6]: Same comments as attrs/msg.rs.

File: packages/mocks/README.md [hunk 9]
```diff
@@ -16,7 +16,7 @@ This crate is part of the provwasm repository, licensed under the Apache License
 
 // ref: contracts/name/src/msg.rs
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
Comment (Taztingo) [inline packages/mocks/README.md:19]: Same comments as attrs/src/msg.rs for query. It should contain the new cw_serde attribute and have attributes for the enum types.

**Review by Taztingo — CHANGES_REQUESTED** [review]
Changes all make sense, nice job! There are two recommendations they made when migrating which are using #[cw_serde] and other attributes for the QueryMsg. Most of the fixes are just adding those and repeating it across each module. The attrs module has my actual comments, and it's just repeated for the others.

File: contracts/attrs/src/msg.rs [hunk 10]
```diff
@@ -30,26 +30,26 @@ pub enum ExecuteMsg {
...
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum QueryMsg {
     GetLabelName {},
     GetLabels {},
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/attrs/Cargo.toml [hunk 11]
```diff
@@ -28,12 +28,12 @@ backtraces = []
...
+cosmwasm-std = { version = "1.1.2", default-features = false }
+cosmwasm-storage = { version = "1.1.2" }
 schemars = "0.8.1"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.20" }
 
 [dev-dependencies]
 provwasm-mocks = { path = "../../packages/mocks" }
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/Cargo.toml]: You are correct, I was delaying the schema refactoring, but is now included with this PR

File: contracts/attrs/src/msg.rs [hunk 12]
```diff
@@ -1,19 +1,19 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/attrs/src/msg.rs [hunk 13]
```diff
@@ -1,19 +1,19 @@
...
 #[serde(rename_all = "snake_case")]
 pub struct InitMsg {
     pub name: String, // Bind this name to the contract address (eg contract.pb).
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/attrs/src/msg.rs [hunk 14]
```diff
@@ -1,19 +1,19 @@
...
 #[serde(rename_all = "snake_case")]
 pub struct Label {
     pub text: String,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/attrs/src/msg.rs [hunk 15]
```diff
@@ -30,26 +30,26 @@ pub enum ExecuteMsg {
     },
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/attrs/src/msg.rs [hunk 16]
```diff
@@ -30,26 +30,26 @@ pub enum ExecuteMsg {
...
 #[serde(rename_all = "snake_case")]
 pub struct LabelNameResponse {
     pub name: String,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/attrs/src/msg.rs [hunk 17]
```diff
@@ -30,26 +30,26 @@ pub enum ExecuteMsg {
...
 #[serde(rename_all = "snake_case")]
 pub struct LabelsResponse {
     pub labels: Vec<Label>,
 }
 
 /// Migrate the contract.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/attrs/src/msg.rs]: now using `cw_serde`

File: contracts/marker/Cargo.toml [hunk 18]
```diff
@@ -29,12 +29,12 @@ backtraces = []
...
+cosmwasm-std = { version = "1.1.2" }
+cosmwasm-storage = { version = "1.1.2" }
 schemars = "0.8.1"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.20" }
 
 [dev-dependencies]
 provwasm-mocks = { path = "../../packages/mocks" }
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/marker/Cargo.toml]: You are correct, I was delaying the schema refactoring, but is now included with this PR

File: contracts/msgfees/Cargo.toml [hunk 19]
```diff
@@ -28,12 +28,12 @@ backtraces = []
...
+cosmwasm-std = { version = "1.1.2" }
+cosmwasm-storage = "1.1.2"
 schemars = "0.8.1"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.20" }
 
 [dev-dependencies]
 provwasm-mocks = { path = "../../packages/mocks" }
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/msgfees/Cargo.toml]: You are correct, I was delaying the schema refactoring, but is now included with this PR

File: contracts/name/Cargo.toml [hunk 20]
```diff
@@ -28,12 +28,12 @@ backtraces = []
...
+cosmwasm-std = { version = "1.1.2" }
+cosmwasm-storage = { version = "1.1.2" }
 schemars = "0.8.1"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.20" }
 
 [dev-dependencies]
 provwasm-mocks = { path = "../../packages/mocks" }
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/name/Cargo.toml]: You are correct, I was delaying the schema refactoring, but is now included with this PR

File: contracts/scope/Cargo.toml [hunk 21]
```diff
@@ -28,13 +28,13 @@ backtraces = []
...
+cosmwasm-storage = { version = "1.1.2" }
 schemars = "0.8.1"
 serde = { version = "1.0.127", default-features = false, features = ["derive"] }
 thiserror = { version = "1.0.20" }
 
 
 [dev-dependencies]
 provwasm-mocks = { path = "../../packages/mocks" }
-cosmwasm-schema = { version = "1.0.0" }
+cosmwasm-schema = { version = "1.1.2" }
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/scope/Cargo.toml]: You are correct, I was delaying the schema refactoring, but is now included with this PR

File: contracts/tutorial/Cargo.toml (hunk 2 above)
Comment (kwtalley) [reply contracts/tutorial/Cargo.toml]: You are correct, I was delaying the schema refactoring, but is now included with this PR

File: contracts/marker/src/msg.rs [hunk 22]
```diff
@@ -2,13 +2,13 @@ use cosmwasm_std::Uint128;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/marker/src/msg.rs]: now using `cw_serde`

File: contracts/msgfees/src/msg.rs [hunk 23]
```diff
@@ -2,14 +2,14 @@ use cosmwasm_std::{Addr, Coin};
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
```
[the remark this answers is not in the document]
Comment (kwtalley) [reply contracts/msgfees/src/msg.rs]: now using `cw_serde`

File: contracts/scope/src/msg.rs (hunk 1 above)
Comment (kwtalley) [reply contracts/scope/src/msg.rs]: now using `cw_serde`

File: contracts/tutorial/src/msg.rs (hunk 3 above)
Comment (kwtalley) [reply contracts/tutorial/src/msg.rs]: now using `cw_serde`

File: packages/bindings/Cargo.toml (hunk 4 above)
Comment (kwtalley) [reply packages/bindings/Cargo.toml]: done

File: packages/bindings/src/msg.rs (hunk 5 above)
Comment (kwtalley) [reply packages/bindings/src/msg.rs:15]: done

File: packages/bindings/src/querier.rs (hunk 6 above)
Comment (kwtalley) [reply packages/bindings/src/querier.rs:157]: the bindings package actually remains unchanged. only inputs/outputs of the contracts use the new derived schema

File: packages/bindings/src/query.rs (hunk 7 above)
Comment (kwtalley) [reply packages/bindings/src/query.rs:8]: the bindings package actually remains unchanged. only inputs/outputs of the contracts use the new derived schema

File: packages/bindings/src/types.rs (hunk 8 above)
Comment (kwtalley) [reply packages/bindings/src/types.rs:6]: the bindings package actually remains unchanged. only inputs/outputs of the contracts use the new derived schema

File: packages/mocks/README.md (hunk 9 above)
Comment (kwtalley) [reply packages/mocks/README.md]: done

**Review by Taztingo — APPROVED** [review]
LGTM

**Pull request merged**